pkg/asset/installconfig/aws/route53.go

```diff
@@ -156,43 +155,25 @@ func GetR53ClientCfg(sess *awss.Session, roleARN string) *aws.Config {
 	return &aws.Config{Credentials: creds}
 }
 
-// CreateOrUpdateRecord Creates or Updates the Route53 Record for the cluster endpoint.
-func (c *Client) CreateOrUpdateRecord(ctx context.Context, ic *types.InstallConfig, target string, intTarget string, phzID string, aliasZoneID string) error {
-	useCNAME := cnameRegions.Has(ic.AWS.Region)
-
-	apiName := fmt.Sprintf("api.%s.", ic.ClusterDomain())
-	apiIntName := fmt.Sprintf("api-int.%s.", ic.ClusterDomain())
-
-	// Create api record in public zone
-	if ic.Publish == types.ExternalPublishingStrategy {
-		zone, err := c.GetBaseDomain(ic.BaseDomain)
-		if err != nil {
-			return err
-		}
-
-		svc := route53.New(c.ssn) // we dont want to assume role here
-		if _, err := createRecord(ctx, svc, aws.StringValue(zone.Id), apiName, target, aliasZoneID, useCNAME); err != nil {
-			return fmt.Errorf("failed to create records for api: %w", err)
-		}
-		logrus.Debugln("Created public API record in public zone")
-	}
-
-	// Create service with assumed role for PHZ
-	svc := route53.New(c.ssn, GetR53ClientCfg(c.ssn, ic.AWS.HostedZoneRole))
+// CreateRecordInput collects information for creating a record.
+type CreateRecordInput struct {
```

Comment:
nit: It might be a good idea to document each field for future use.

Reply:
Anything unclear about the field names that we could fix/rename instead of adding comments?

Reply:
I don't think so. They seem pretty good. I just thought some additional comments might help when coming back to it in the future.

Reply:
Fair enough, I'll add some comments.

Reply:
Done.